Dockerfile

```diff
@@ -0,0 +1,46 @@
+# Use Arch Linux as the base image
+FROM archlinux:latest AS builder
+
+# Install necessary dependencies
+RUN pacman -Syu --noconfirm \
+    base-devel \
+    git \
+    go \
+    libvips \
+    ffmpeg \
+    libheif
+
+# Set the working directory inside the container
+WORKDIR /app
+COPY . .
+
+ENV CGO_ENABLED=1
+RUN go env -w GOCACHE=/go/cache
+RUN go env -w GOMODCACHE=/go/modcache
+RUN --mount=type=cache,target=/go/modcache go mod download
+RUN --mount=type=cache,target=/go/modcache --mount=type=cache,target=/go/cache go build -o /app/timelinize
+
+FROM archlinux:latest AS final
+
+WORKDIR /app
+
+RUN pacman -Syu --noconfirm \
+    libvips \
+    ffmpeg \
+    libheif
+
+RUN useradd -u 1000 -m -s /bin/bash -d /app timelinize
+RUN mkdir -p /app/.config/timelinize /repo
+RUN chown -R timelinize /app
+RUN chown -R timelinize /repo
+
+COPY --from=builder /app/timelinize /app/timelinize
+
+ENV TIMELINIZE_ADMIN_ADDR="0.0.0.0:12002"
```

Comment:
Normally I'd say this is a bad idea (for obvious reasons, leaving a trusted API open on a wildcard interface) but in a Docker container I guess the only thing that can get into it is what you forward from the host.

Reply:
Yeah, I'm not a Docker expert, but this is fairly common I think, if you want your service to be exposed using the Docker `--publish` flag. Without the `--publish` flag, the service address wouldn't be reachable even if it listens in all interfaces.